### PR TITLE
fix docker in docker canary jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
- disable git cache for verify
- fix volume mounts on prow
- strip newline in .git """symlink"""

/area scenarios
/area jobs

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -3275,7 +3275,6 @@ presubmits:
         imagePullPolicy: Always
         args:
         - "--clean"
-        - "--git-cache=/root/.cache/git"
```
Comment (krzyzacy) [inline prow/config.yaml]: (this is in `pull-security-kubernetes-verify-prow`)

Comment (BenTheElder) [reply prow/config.yaml]: fixed

Comment (krzyzacy) [reply prow/config.yaml]: and assume we don't need to mount the volume in the job anymore

Comment (BenTheElder) [reply prow/config.yaml]: the volume is used for docker image caching :-)

Comment (BenTheElder) [reply prow/config.yaml]: discussed out of band, updated volumes a bit more.

**Pull request merged**